packages/web/src/components/track/ViewerActionButtons.tsx

```diff
@@ -0,0 +1,164 @@
+  const { data: track } = useGetTrackById({ id: contentId })
+  const { data: collection } = useGetPlaylistById({ playlistId: contentId })
+
+  const {
+    stream_conditions: streamConditions,
+    has_current_user_saved: isFavorited,
+    has_current_user_reposted: isReposted
+  } = (type === 'track' ? track : collection) ?? {}
+
+  const isUnlisted =
+    type === 'track' ? track?.is_unlisted : collection?.is_private
+  return {
+    streamConditions,
+    isUnlisted,
+    isFavorited,
+    isReposted
+  }
+}
+
+export const ViewerActionButtons = ({
+  contentId,
+  contentType,
+  hasStreamAccess,
+  isDisabled,
+  isLoading,
+  rightActions,
+  bottomBar,
+  isDarkMode,
+  isMatrixMode,
+  showIconButtons,
+  onClickRepost,
+  onClickFavorite,
+  onClickShare,
+  onClickGatedUnlockPill
+}: ViewerActionButtonProps) => {
+  const { streamConditions, isUnlisted, isFavorited, isReposted } =
+    useEntityDetails(contentId, contentType)
+
+  const gatedStatusMap = useSelector(getGatedContentStatusMap)
+  const gatedStatus = contentId && gatedStatusMap[contentId]
+
+  const repostLabel = isReposted ? messages.unrepost : messages.repost
+
+  const onStopPropagation = useCallback((e: any) => e.stopPropagation(), [])
+
+  if (streamConditions && !isLoading && !hasStreamAccess) {
+    return (
+      <Flex justifyContent='space-between' w='100%' alignItems='center'>
+        <Text variant='title' size='s'>
+          <GatedConditionsPill
+            streamConditions={streamConditions}
+            unlocking={gatedStatus === 'UNLOCKING'}
+            onClick={onClickGatedUnlockPill}
+          />
+        </Text>
+        <Flex>{rightActions}</Flex>
+      </Flex>
+    )
+  }
+
+  return (
+    <Flex justifyContent='space-between' w='100%' alignItems='center'>
+      {bottomBar}
+      {!isLoading && showIconButtons && !isUnlisted ? (
+        <Flex gap='2xl'>
+          <Tooltip
+            text={repostLabel}
+            disabled={isDisabled}
+            placement='top'
+            mount='page'
+          >
+            <Flex css={{ position: 'relative' }}>
+              <RepostButton
+                aria-label={repostLabel}
+                onClick={onClickRepost}
+                isDisabled={isDisabled}
+                isActive={isReposted}
+                isDarkMode={!!isDarkMode}
+                isMatrixMode={isMatrixMode}
+              />
+            </Flex>
+          </Tooltip>
+          <Tooltip
+            text={isFavorited ? messages.unfavorite : messages.favorite}
+            disabled={isDisabled}
+            placement='top'
+            mount='page'
+          >
+            <Flex css={{ position: 'relative' }}>
+              <FavoriteButton
+                onClick={onClickFavorite}
+                isActive={isFavorited}
+                isDisabled={isDisabled}
+                isDarkMode={!!isDarkMode}
+                isMatrixMode={isMatrixMode}
```

Comment:
do these other buttons also need an `aria-label`?

Reply:
Hmm looks like they're a little jank to be completely honest. I added aria-label on the inside of the component. Seems like a good candidate for refactor but I don't want to include that scope at the moment 😅